File: backend/app/api/routes/system.py
"""文件功能：提供平台级公开启动配置与系统默认值，供前端初始化时读取。"""

from fastapi import APIRouter

from app.core.config import get_settings
from app.schemas.project_app_config import DEFAULT_PROJECT_STYLE_SPEC_MARKDOWN

router = APIRouter()


@router.get("/system/settings")
async def get_system_settings() -> dict[str, str]:
    """公开前端启动所需的业务时区，不返回凭证或其它服务端配置。"""

    return {"app_timezone": get_settings().app_timezone}


@router.get("/system/default-style-spec")
async def get_default_style_spec() -> dict[str, str]:
    """返回平台级默认 Markdown 样式规范，作为前端表单初始值的统一来源。"""

    return {"style_spec_markdown": DEFAULT_PROJECT_STYLE_SPEC_MARKDOWN}

File: editor/src/api/system.ts
/**
 * 文件功能：封装平台级系统默认值的只读查询接口。
 */
import { http } from '@/api/http'

export interface DefaultStyleSpec {
  style_spec_markdown: string
}

export interface SystemSettings {
  app_timezone: string
}

/** 读取后端业务时区，使已构建的前端也能跟随部署配置。 */
export async function fetchSystemSettings(): Promise<SystemSettings> {
  const { data } = await http.get<SystemSettings>('/system/settings')
  return data
}

/** 获取平台级默认 Markdown 样式规范，供前端表单初始值使用。 */
export async function fetchDefaultStyleSpec(): Promise<DefaultStyleSpec> {
  const { data } = await http.get<DefaultStyleSpec>('/system/default-style-spec')
  return data
}
